Harvard Caselaw Access Project.js

```diff
@@ -0,0 +1,175 @@
+{
+	"translatorID": "2a1cafb9-6f61-48d3-b621-c3265fde9eba",
+	"label": "Harvard Caselaw Access Project",
+	"creator": "Franklin Pezzuti Dyer",
+	"target": "^https://(.*\\.)*case\\.law",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2024-01-15 23:41:24"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2024 Franklin Pezzuti Dyer
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, _url) {
+	if (doc.querySelectorAll('.case-container').length > 0) {
+		return 'case';
+	}
+	else if (getSearchResults(doc, true)) {
+		return 'multiple';
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('div.result-title > div > a');
+	for (let row of rows) {
+		let href = row.href;
+		Z.debug(href);
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, false));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url = doc.location.href) {
+	let apiUrl = attr(doc, "a[href*='api.case.law/v1/cases/']", 'href');
+	let caseText = await requestText(apiUrl);
+	let caseJson = JSON.parse(caseText);
+
+	let caseItem = new Zotero.Item("case");
+	caseItem.language = "en-US";
+	caseItem.url = url;
+
+	caseItem.caseName = caseJson.name;
+	caseItem.shortTitle = caseJson.name_abbreviation;
+	caseItem.court = caseJson.court.name;
+	caseItem.dateDecided = caseJson.decision_date;
+	caseItem.docketNumber = caseJson.docket_number;
+	caseItem.reporter = caseJson.reporter.full_name;
+	caseItem.reporterVolume = caseJson.volume.volume_number;
+
```

Comment:
I'm capturing the parallel citation in `history` in the Court Listener translator and I think that makes sense. They're often required in citations and while we can't 100% do this with Zotero, we can at least approximate it. You could iterate over citations.cite and concat with a comma, then write to history.